src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -155,8 +155,6 @@ def _check_pandas_version(python_range, current_pandas_version, pandas_version_r
     if args.fields:
         if args.sql:
             file_name_suffix = '.data.sql'
-        elif resp["downloadRestricted"]:
```

Comment:
Looks like we can just remove this elif part.
But when I tested with that change, it eventually started connecting to thrift and fails there:
oot@job-GKKFPQk2FVxJ81qJPXP641jy:/# dx extract_dataset project-GK35pk92FVx10Q9fKQg4kqgK:record-GK9Gf0k2FVxP3V2G8Zv0xv8b --fields participant.pid
=========== downloadRestricted is true
{'type': 'ThreadPoolException', 'message': 'Failed to fetch data', 'details': "'NoneType' object has no attribute 'cursor'", 'time': '12/16/2022, 22:20:15'}

But that looks like different issue, related to the signature verification. So this change looks fine